packages/ui/src/WalletModal.tsx

```diff
@@ -129,6 +129,12 @@ export const WalletModal: FC<WalletModalProps> = ({ className = '', container =
 
     useLayoutEffect(() => setPortal(document.querySelector(container)), [container]);
 
+    useLayoutEffect(() => {
+        if (wallet) {
+            connect(decryptPermission || "NO_DECRYPT", network || WalletAdapterNetwork.TestnetBeta, programs ?? []).catch((e) => {console.log({e})});
```

Comment:
I'm not sure how this detects properly if we're on mobile. Can we instead use something like: https://www.npmjs.com/package/react-device-detect ?